### PR TITLE
Feature/gh 7953

### LINKED ISSUE #7953: Create zoom image component and thumbnails
This component allows to see a larger view of the product image and to zoom on it. The component also shows thumbnails images and allows to cycle through the various product view.

### Acceptance

- Moving mouse scrolls image left right (arrows)
- Clicking zooms x2
- Thumbnails are visible and allow to switch the "featured" image
- **Touch interactions to be done in follow up ticket**

### PR DESCRIPTION
**Pull request by LTiger14**
Closes #7953

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts [hunk 1]
```diff
@@ -10,6 +11,7 @@ import { CurrentProductService } from '../current-product.service';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ProductImagesComponent {
+  iconTypes = ICON_TYPE;
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts]: Where this `iconTypes` is used?

File: projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts [hunk 2]
```diff
@@ -30,10 +32,15 @@ export class ProductImagesComponent {
     map(([, container]) => container)
   );
 
+  // Used for image zoom
+  expandImage = new BehaviorSubject(false);
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts:36]: `expendImage` is confusing. It is not `image` but whether the image is expended, so may be `isImageExpended$` or `isImageDialogOpened$` or `isImageZoomed$`... or whatever make sense :)

File: projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts (hunk 1 above)
Comment (LTiger14) [reply projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts]: My bad, removed

Comment (LTiger14) [reply projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts]: Removed

File: projects/storefrontlib/src/shared/components/image-zoom/product-view/image-zoom-product-view.component.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+<ng-container *ngIf="thumbs$ | async as thumbs">
+  <cx-carousel
+    *ngIf="thumbs.length"
+    class="thumbs"
+    [items]="thumbs"
+    itemWidth="70px"
+    [hideIndicators]="false"
+    [template]="thumb"
+    [hideIndicators]="true"
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/shared/components/image-zoom/product-view/image-zoom-product-view.component.html:28]: why there are 2 `hideIndicators`?

File: projects/storefrontlib/src/shared/components/image-zoom/dialog/image-zoom-dialog.component.ts [hunk 4]
```diff
@@ -17,7 +17,20 @@ export class ImageZoomDialogComponent {
     focusOnEscape: true,
   };
 
-  constructor(protected launchDialogService: LaunchDialogService) {}
+  @Input() galleryItem: number;
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/shared/components/image-zoom/dialog/image-zoom-dialog.component.ts:20]: Can we change `galleryItem` to `galleryIndex`?

**Pull request closed without merging**